python_benchmarks/qr_codes/qr.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+
+"""qr.py
+Usage example:
+python qr.py -o out.yaml -p qrcodes/detection
+-H, --help - show help
+-o, --output - output file (default out.yaml)
+-p, --path - input dataset path (default qrcodes/detection)
+-a, --accuracy - input accuracy (default 47)
+-alg - input alg (default OpenCV)
+-m, --metric - input metric (default ~)
+"""
+
+import argparse
+import glob
+
+import cv2 as cv
+import numpy as np
+
+
+def init_qr_detector_open_cv():
+    return cv.QRCodeDetector()
+
+
+def init_wechat_qr_detector_open_cv(path_to_model):
+    return cv.wechat_qrcode.WeChatQRCode(path_to_model+'/detect.prototxt', path_to_model+'/detect.caffemodel',
+                                         path_to_model+'sr.caffemodel')
+
+
+def detect_qr_opencv(image_path, detector):
+    image = cv.imread(image_path, cv.IMREAD_IGNORE_ORIENTATION)
+    #ret, corners = detector.detect(image)
+    ret, corners = detector.detectMulti(image)
+    return ret, corners
+
+
+def detect_wechat_qr_opencv(image_path, detector):
+    image = cv.imread(image_path, cv.IMREAD_IGNORE_ORIENTATION)
+    ret, corners = detector.detectAndDecode(image)
+    return ret, corners
+
+
+def find_images_path(dir_path):
+    images = glob.glob(dir_path+'/*.jpg')
+    images += glob.glob(dir_path+'/*.png')
+    return images
+
+
+def get_corners(label_path):
+    f = open(label_path, "r")
+    corners = []
+    for line in f.readlines():
+        try:
+            f_list = [float(i) for i in line.split(" ")]
+            corners += f_list
+        except:
+            pass
+    return np.array(corners).reshape(-1, 4, 2)
+
+
+def get_distance(gold_corners, corners):
+    return abs(np.amax(gold_corners - corners))
+
+
+def get_distance_to_rotate_qr(gold_corner, corners, accuracy):
+    corners = corners.reshape(-1, 4, 2)
+    dist = 1e9
+    for one_corners in corners:
+        dist = get_distance(gold_corner, one_corners)
+        if dist > accuracy:
+            for i in range(0, 3):
+                if dist > accuracy:
+                    one_corners = np.roll(one_corners, 1, 0)
+                    dist = min(dist, get_distance(gold_corner, one_corners))
+                else:
+                    return dist
+        if dist > accuracy:
+            one_corners = np.flip(one_corners, 0)
+            dist = min(dist, get_distance(gold_corner, one_corners))
+            for i in range(0, 3):
+                if dist > accuracy:
+                    one_corners = np.roll(one_corners, 1, 0)
+                    dist = min(dist, get_distance(gold_corner, one_corners))
+                else:
+                    return dist
+    return dist
```

Comment:
It should be max distance, but not the last one in the loop.

Reply:
`get_norm_to_rotate_qr` finds the smallest distance between the gold QR and all possible rotations of the found QR code.